### PR TITLE
Shxco apache to nginx upgrade

### PR DESCRIPTION
**Pull request by quadrismegistus**
For https://github.com/Princeton-CDH/cdh-ansible/issues/148

also includes a fix for #143 and changes for #169

### CODE REVIEW

**Comment by quadrismegistus** [comment]
This error seems to be arising from issues with overwriting the python environment:

```
TASK [python : Upgrade pip/setuptools] ************************************************************************
fatal: [cdh-test-shxco1.princeton.edu]: FAILED! => changed=false 
  cmd:
  - /srv/www/mep/1.6.0-dev-c9f08c/env/bin/pip3
  - install
  - -U
  - pip
  - setuptools
  msg: |-
    stdout: Requirement already satisfied: pip in /srv/www/mep/1.6.0-dev-c9f08c/env/lib/python3.8/site-packages (23.3)
    Requirement already satisfied: setuptools in /srv/www/mep/1.6.0-dev-c9f08c/env/lib/python3.8/site-packages (57.5.0)
    Collecting setuptools
      Using cached setuptools-68.2.2-py3-none-any.whl.metadata (6.3 kB)
    Using cached setuptools-68.2.2-py3-none-any.whl (807 kB)
    Installing collected packages: setuptools
      Attempting uninstall: setuptools
        Found existing installation: setuptools 57.5.0
        Uninstalling setuptools-57.5.0:
  
    :stderr: ERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: 'setup.cpython-38.pyc'
    Check the permissions.

PLAY RECAP ****************************************************************************************************
cdh-test-shxco1.princeton.edu : ok=50   changed=7    unreachable=0    failed=1    skipped=5    rescued=0    ignored=0   
lib-solr-staging4.princeton.edu : ok=12   changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   
```

File: group_vars/shxco_qa/vars.yml [hunk 1]
```diff
@@ -12,6 +12,11 @@ zk_host: "lib-zk-staging1:2181,lib-zk-staging2:2181,lib-zk-staging3:2181/solr8"
 solr_url: "http://lib-solr8-staging.princeton.edu:8983/solr/"
 solr_server: "{{ groups['solr_staging'][0] }}"
 
+# set passenger defaults for production; override for other environments
+passenger_app_root: "/var/www/{{ app_name }}"
+passenger_server_name: "shakespeareandco.princeton.edu"
+passenger_startup_file: "{{ app_name }}/wsgi.py"
+passenger_python: "{{ passenger_app_root }}/env/bin/python"
```
Comment (rlskoeser) [inline group_vars/shxco_qa/vars.yml:19]: these should be set in the main shxco vars and then qa vars should override the passenger server name with the qa hostname https://test-shakespeareandco.cdh.princeton.edu/

**Review by rlskoeser — APPROVED** [review]
Changes look good. I think one set of variables should be moved/tweaked slightly to share and differentiate qa and prod. 

Make sure to remove the commented out apache stuff before merging.

File: playbooks/shxco_qa.yml [hunk 2]
```diff
@@ -18,12 +18,13 @@
...
     - build_project_repo
     - postgresql
+    - passenger
     - build_npm
     - run_webpack   # dependency for collectstatic
     - configure_logging # logging directory must exist before running django commands
     - django
     - solr_collection
-    - configure_apache
+    # - configure_apache
```
Comment (rlskoeser) [inline playbooks/shxco_qa.yml:27]:
```suggestion
```

**Review by rlskoeser — APPROVED** [review]

File: roles/build_npm/tasks/main.yml [hunk 3]
```diff
@@ -19,14 +19,17 @@
...
-    # https://serverfault.com/a/1025300
+    # NOTE: upgrading nodejs with ansible snap fails, even though
+    # the documentation claims it should refresh when then channel changes.
+    # Manually run a refresh command to ensure version changes take effect, e.g.:
+    #   sudo snap refresh node --channel=18
+    # NOTE2: could add a node -v check and only refresh on mismatch
 
+    - name: Refresh nodejs to ensure version changes take effect
+      become: true
+      ansible.builtin.command: "snap refresh node --channel={{ node_version }}/stable"
```
Comment (rlskoeser) [inline roles/build_npm/tasks/main.yml:32]: @kayiwa FYI, we (@quadrismegistus and I) needed to solve the node upgrade problem (#143) and came up with a simpler solution than switching to the princeton ansible role (as previously proposed). Running the refresh command here resolves the problem, and doesn't seem to cause any problems or delay when refresh has been done. The comment above should make it clear why this is here, and if/when ansible actually runs the refresh as it claims to do this stanza can be removed.

**Pull request merged**